lib/DXIL/DxilResourceBase.cpp

```diff
@@ -43,7 +45,13 @@ unsigned DxilResourceBase::GetUpperBound() const  { return m_RangeSize != UINT_M
 unsigned DxilResourceBase::GetRangeSize() const   { return m_RangeSize; }
 llvm::Constant *DxilResourceBase::GetGlobalSymbol() const { return m_pSymbol; }
 const std::string &DxilResourceBase::GetGlobalName() const      { return m_Name; }
-llvm::Value *DxilResourceBase::GetHandle() const  { return m_pHandle; }
+llvm::Value *DxilResourceBase::GetHandle() const { return m_pHandle; }
+// If m_pHLSLTy is nullptr, HLSL type is the type of m_pSymbol.
+// In sm6.6, type of m_pSymbol will be mutated to handleTy, m_pHLSLTy will save
+// the original HLSL type.
+llvm::Type *DxilResourceBase::GetHLSLType() const {
+  return m_pHLSLTy == nullptr ? m_pSymbol->getType() : m_pHLSLTy;
```

Comment:
Should we return nullptr if m_pSymbol is nullptr?

Reply:
Will cover this in another PR.